ops/c/include/ops_internal2.h

```diff
@@ -287,6 +287,20 @@ void ops_put_data(ops_dat dat);
 OPS_FTN_INTEROP
 void create_kerneldesc_and_enque(char const *name, ops_arg *args, int nargs, int index, int dim, int isdevice, int *range, ops_block block, void (*func)(struct ops_kernel_descriptor *desc));
 
+
+/*******************************************************************************
+* Random number generations
+*******************************************************************************/
+void ops_randomgen_init(unsigned int seed, int options);
+void ops_fill_random_uniform(ops_dat dat);
+void ops_fill_random_normal(ops_dat dat);
+void ops_randomgen_exit();
+
+void ops_randomgen_init_host(unsigned int seed, int options, std::mt19937 ops_rand_gen);
```

Comment:
What is the reason for passing ops_rand_gen through? Especially if it's fixed to having std::mt19937 as a type?

Reply:
The host is common implementation among host, sycl and opencl

so in sycl_common, host_common and opencl_common we have ops_randomgen_init which will call this implementation at backend as this similar for all
thats why passing here the global delcared in those files here for initilization

Reply:
I understand why we call to ops_randomgen_init_host - my issue is why have the ops_rand_gen global three times in the three backends, when its only use is in ops_lib_core.cpp. Why not there?

Reply:
yes as per your suggestion, i moved common implementation to ops_lib_core.cpp which this sycl, host and opencl will call

should i move the declarations also from ops_internal2.h to ops_lib_core.h

Reply:
No, those are internal functions the use should not be calling directly, so they belong in ops_internal2.h